lib/model/indexhandler.go

```diff
@@ -348,6 +442,24 @@ func (s *indexHandler) receive(fs []protocol.FileInfo, update bool, op string) e
 	return nil
 }
 
+var warnSequenceAnomalyOnce sync.Once
+
+func (s *indexHandler) logSequenceAnomaly(msg string, extra map[string]any) {
+	warnSequenceAnomalyOnce.Do(func() {
+		l.Warnf("Index sequence anomaly detected (please report at https://forum.syncthing.net/t/22660): %s (%v)", msg, extra)
+	})
+
+	extraStrs := make(map[string]string, len(extra))
+	for k, v := range extra {
+		extraStrs[k] = fmt.Sprint(v)
+	}
+
+	s.evLogger.Log(events.Failure, ur.FailureData{
+		Description: msg,
+		Extra:       extraStrs,
+	})
+}
+
```

Comment:
Could have logged to sentry?

Reply:
Indeed it does, via the failure reporting mechanism 👍